team-telnyx/telnyx-webrtc-ios
# WEBRTC-420: Add SDK support for ringback and ringtone.

[WebRTC-420 - [iOS] Add SDK support for ringback and ringtone](https://telnyx.atlassian.net/browse/WEBRTC-420)

---


- SDK Ringtone and Ringback tone support.
- Custom Ringtone and Ringback tone support. 
- Demo App setting a custom Ringtone and Ringback tone.

## :older_man: :baby: Behaviors

### Before changes
- The SDK did not support to set a custom ringtone and ringback tone.

### After changes
- SDK supports setting a custom ringtone and ringback tone.
- Added a custom ringtone and ringback tone to the demo app. 

## ✋ Manual testing

1. Build the WebRTCSDK project. 
2. Build and run the demo app project.
3. Login with a SIP user. 
4. Place a call from the app: The ringback tone should be heard until the the call is rejected / answered.
5. Receive an call: The ringtone should be heard until the call is rejected / answered.

**To change the ringtone / ringback tone:** 
1. Add the audio files to the demo app project. 
2. Change the initial the ringtone / ringback tone parameters from the TxConfig: 

```
let txConfig = TxConfig(sipUser: sipUser,
                         password: password,
                         ringtone: "incoming_call.mp3",
                         ringBackTone: "ringback_tone.mp3")

 self.telnyxClient?.connect(txConfig: txConfig)
```

Notice that **ringtone** and **ringBackTone** are not mandatory fields and can be omitted: 

```
let txConfig = TxConfig(sipUser: sipUser,
                        password: password,
                        ringtone: "incoming_call.mp3")

 self.telnyxClient?.connect(txConfig: txConfig)
```

Or login in with token:

```
let txConfig = TxConfig(token: token,
                         ringtone: "incoming_call.mp3")

 self.telnyxClient?.connect(txConfig: txConfig)
```
